woocommerce/woocommerce-ios
# Update top performers content - cell configuration, section header, and items sorted by items sold

Closes: #5793 


### Description


This PR updates the content of the top performers section in the My Store tab. Previously (production version), the top-performing products are sorted by "total amount" in a hacky way due to API limitation (p91TBi-716-p2). With the new design  to better match core, the products are now sorted by "items sold" as returned in the Leaderboards API. The section header labels and cell configuration are also updated based on the latest design.

Major changes include:

- Updated `ProductTableViewCell` for new cell configuration and refactored with a view model for unit testing
  - The view model can be initialized with a `TopEarnerStatsItem` and the feature flag value
  - Its `priceLabel` property is renamed to `accessoryLabel` for flexibility
  - Added unit tests for the view model when the feature flag is on/off
- Updated the top performers section labels in `TopPerformerDataViewController`
- Updated how top-performing products are sorted by updating its `Comparable` implementation (unfortunately we can't feature flag this, we will manually test the production version)
  - Added unit tests for comparing `TopEarnerStatsItem`
  - Implemented `Copiable` for `TopEarnerStatsItem` for easier unit testing

### Testing instructions


Prerequisite: the store has at least two orders in any of the time range (today/this week/this month/this year)

- Launch the app
- After the stats are synced, tap each time range tab --> if the time range has any orders, the top performers section should show the updated header labels ("Products" and "Items Sold") and cell content. The products should be ordered by quantity instead of total amount like before

### Screenshots


<img width="826" alt="Screen Shot 2022-01-04 at 3 02 26 PM" src="https://user-images.githubusercontent.com/1945542/148025378-cd706e3c-bfb1-4c6c-bda5-aef7b3c72f1a.png">

\ | before | after
-- | -- | --
dark | ![Simulator Screen Shot - iPhone 11 - 2022-01-04 at 15 43 15](https://user-images.githubusercontent.com/1945542/148025667-c17624db-4dfb-4813-8e2b-ae5ab26c0251.png) | ![Simulator Screen Shot - iPhone 11 - 2022-01-04 at 15 04 19](https://user-images.githubusercontent.com/1945542/148025418-3d9ceb0e-24c0-48c1-8ade-89415fee2c1f.png)
light | ![Simulator Screen Shot - iPhone 11 - 2022-01-04 at 15 42 53](https://user-images.githubusercontent.com/1945542/148025657-e67fceef-78b3-4b14-9eff-a607e391d661.png) | ![Simulator Screen Shot - iPhone 11 - 2022-01-04 at 15 04 34](https://user-images.githubusercontent.com/1945542/148025422-eca14a69-93f9-4a9f-a9ca-c25cee54a19b.png)



---
